src/realm/sync/network/http.hpp

```diff
@@ -306,6 +309,15 @@ struct HTTPParser : protected HTTPParserBase {
         m_socket.async_read_until(m_read_buffer.get(), max_header_line_length, '\n', std::move(handler));
     }
 
+    inline uint32_t hex_to_int(const std::string& hex) throw()
+    {
+        uint32_t dec;
+        std::stringstream ss;
+        ss << std::hex << hex;
```

Comment:
nit: This could have been `std::strtoul` which will be significantly more performant than using a stringstream.